lib/elastic_apm/spies/mongo.rb

```diff
@@ -96,9 +96,8 @@ def push_event(event)
 
         def pop_event(event)
           span = events.delete(event.operation_id)
```

Comment:
If the agent is not recording, will this not be a problem? We have an early return in `push_event` if there's no transaction. Shouldn't we have the same check here?